utils.go

```diff
@@ -2,19 +2,26 @@ package gtrs
 
 import (
 	"context"
+	"encoding/json"
 	"errors"
 	"fmt"
 	"reflect"
 	"regexp"
 	"strconv"
 	"strings"
+	"time"
 
 	"github.com/redis/go-redis/v9"
 )
 
+// Metadata is a type that allows serialization of generic structured
+// metadata within the stream entries. Any value that can be serialized
+// to JSON can be inserted here.
+type Metadata map[string]any
```

Comment:
Cool idea. Maybe add a line about it in the readme?

Reply:
Sounds good, added a README section for this 👍